google-cloud-pubsublite/src/main/java/com/google/cloud/pubsublite/internal/CheckedApiException.java

```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.cloud.pubsublite.internal;
+
+import com.google.api.gax.rpc.ApiException;
+import com.google.api.gax.rpc.StatusCode;
+import javax.annotation.Nullable;
+
+public class CheckedApiException extends Exception {
```

Comment:
Nit: can you add a comment on why we need this? IIUC, this just wraps an ApiException but makes it checked for visibility.

Reply:
We don't 'need' this. Its to help with internal correctness, so that we have to handle errors that might arise. IMO, having ApiException inherit from RuntimeException is a mistake, but its what we have to work with.

Reply:
Done.